sdk/storage/azure-storage-files-datalake/src/datalake_file_client.cpp

```diff
@@ -424,7 +424,8 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     protocolLayerOptions.ExpiryOrigin = expiryOrigin;
     if (options.ExpiresOn.HasValue() && options.TimeToExpireInMs.HasValue())
     {
-      throw std::runtime_error("ExpiresOn and TimeToExpireInMs should be mutually exlusive.");
+      // ExpiresOn and TimeToExpireInMs should be mutually exlusive.
+      std::abort();
```

Comment:
I wish there was way to convey this error message string to the caller, similar an exception.

Reply:
There is unfortunately not, that's why the comment here is important so that customers can see this line crashes and in the nearest location they see why.